File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;
    
    // 시작부터 끝까지 차례대로 구하기
    for(long long i = left; i <= right; i++){
        if(i - (i/n) <= (i/n)*n) answer.push_back(i/n+1);
        else answer.push_back(i % n + 1);
        
        
    }
    return answer;
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
import java.util.*; 

class Solution {
    
    public int[] solution(int n, long left, long right) {
        List<Integer> list = new ArrayList<>(); 
        
        // left ~ right까지 수 알아내기
        for(long id = left; id <= right ; id++){ // 10^5
            long row = id / n + 1; 
            long col = id % n + 1; 
            int num = (int)Math.max(row, col); 
            list.add(num); 
        }
        
        return list.stream().mapToInt(i->i).toArray();
    }
}